src/redis-cli.c

```diff
@@ -9835,6 +9918,583 @@ void testHintSuite(char *filename) {
+    if (hdr_init(1, INT64_C(1ULL*1024*1024*1024*1024), 4, &keysize_histogram)) {
+	    fprintf(stderr, "Keystats hdr init error\n");
+	    exit(1);
+    }
+
+    signal(SIGINT, longStatLoopModeStop);
+
+    /* Total keys pre scanning */
+    total_keys = getDbSize();
+
+    /* Status message */
+    printf("\n# Scanning the entire keyspace to find the biggest keys and distribution information.\n");
+    printf("# Use -i 0.1 to sleep 0.1 sec per 100 SCAN commands (not usually needed).\n");
+    printf("# Use --cursor <n> to start the scan at the cursor <n> (usually after a Ctrl-C).\n");
+    printf("# Use --top <n> to display <n> top key sizes (default is 10).\n");
+    printf("# Ctrl-C to stop the scan.\n\n");
+
+    /* Use readonly in cluster */
+    sendReadOnly();
+
+    /* SCAN loop */
+    do {
+        /* Grab some keys and point to the keys array */
+        reply = sendScan(&it);
+        scan_loops++;
+        keys = reply->element[1];
+
+        /* Reallocate our type and size array if we need to */
+        if (keys->elements > array_size) {
+            memkeys_types = zrealloc(memkeys_types, sizeof(typeinfo*)*keys->elements);
+            memkeys_sizes = zrealloc(memkeys_sizes, sizeof(unsigned long long)*keys->elements);
+
+            bigkeys_types = zrealloc(bigkeys_types, sizeof(typeinfo*)*keys->elements);
+            bigkeys_sizes = zrealloc(bigkeys_sizes, sizeof(unsigned long long)*keys->elements);
+
+            if (!memkeys_types || !memkeys_sizes || !bigkeys_types || !bigkeys_sizes) {
+                fprintf(stderr, "Failed to allocate storage for keys!\n");
+                exit(1);
+            }
+
+            array_size = keys->elements;
+        }
+
+        /* Retrieve types and sizes for memkeys */
+        getKeyTypes(memkeys_types_dict, keys, memkeys_types);
+        getKeySizes(keys, memkeys_types, memkeys_sizes, 1, memkeys_samples);
+
+        /* Retrieve types and sizes for bigkeys */
+        getKeyTypes(bigkeys_types_dict, keys, bigkeys_types);
+        getKeySizes(keys, bigkeys_types, bigkeys_sizes, 0, memkeys_samples);
+
+        for(i=0;i<keys->elements;i++) {
+            /* Skip keys that disappeared between SCAN and TYPE */
+            if (!memkeys_types[i] || !bigkeys_types[i]) {
+                continue;
+            }
+
+            total_size += memkeys_sizes[i];
+            sampled++;
+
+            if (updateTopSizes(keys->element[i]->str, keys->element[i]->len, memkeys_sizes[i],
+                               memkeys_types[i]->name, top_sizes, top_sizes_limit) == -1) {
+                fprintf(stderr, "Failed to allocate memory for list node.\n");
+                exit(1);
+            }
+            updateKeyType(keys->element[i], memkeys_sizes[i], memkeys_types[i]);
+            updateKeyType(keys->element[i], bigkeys_sizes[i], bigkeys_types[i]);
+
+            /* Key Size distribution */
+            if (hdr_record_value(keysize_histogram, memkeys_sizes[i]) == 0) {
```

Comment:
I'm wondering if we need this check. I see that other hdr_record_value calls actually don't check the return value.